KOPS.md

```diff
@@ -0,0 +1,130 @@
+
+## Bootstrap AWS encryption provider using kops during cluster creation
+This guide is an extention to bootstrap instruction on README.md
+
+#### Set the `--encryption-provider` flag with kops
+To set the kubernetes `--encryption-provider` flag with kops you must add it to the kops cluster specificition.
+```yaml
+kind: Cluster
+spec:
+  encryptionConfig: true
+
+```
+Also, you need let kops know about the encryption configuration file. To do this you have to create a kops secret with a special `encryptionconfig` flag during cluster creation, in the following order:
+```bash
+kops create -f <path-to-cluster-spec> --state <state-store>
+kops create secret encryptionconfig -f <path-to-encryption-config-file> --state <state-store> --name <cluster-name>
+kops update cluster <cluster-name> --state <state-store> --yes
+```
+#### Permissions
+Ensure master IAM role has permissions to encrypt/decrypt using the kms. You can achieve this
+using additionalIAMPolicies functionality of kops.
+Example:
+```yaml
+kind: Cluster
+spec:
+  additionalPolicies:
+    master: |
+      [
+        {
+          "Effect": "Allow",
+          "Action": [
+            "kms:Decrypt",
+            "kms:Encrypt"
+          ],
+          "Resource": [
+            <arn-of-kms-key>
+          ]
+        }
+      ]
+```
+#### Use Host Network for aws-encryption-provider
+As the CNI plugin is not yet available, you need to add `hostNetwork: true` to pod spec.
+
+#### Update health port for aws-encryption-provider
+When using hostNetwork, the port `8080` used by aws-encryption-provider conflicts with
+kube-apiserver which also requires the same port. To fix this, add `-health-port=:8083`
+to args section of pod spec above. Also change the port in `containerPort` and `livenessProbe`
+sections.
+
+#### Add /var/run/kmsplugin hostMount to api server spec
+Use kops lifecycle hook to run a script/container that can update the kube-apiserver 
+manifest (available at /etc/kubernetes/manifests) to add `/var/run/kmsplugin` as hostMount. Alternatively, you can mount a 
+directory that is already mounted by default e.g. `/srv/kubernetes/socket.sock`.
+
+#### Run aws-encryption-provider as static pod
```

Comment:
Do you want to move this up higher in the guide?

Reply:
Does this still apply after:

> Changed this to To fix this, add `-health-port=:8083` to args section like the pod specification below. Since there is a pod spec near the end of this guide.

In the above conversation?